### ISSUE TITLE
IncompatibleAssign: Explicit cast between string pointer to enum-backing type pointer required

### ISSUE THREAD

**Issue by ambientlight** [Azure/autorest.go#651]
@jhendrixMSFT: Follow-up to https://github.com/Azure/autorest.go/issues/638

**Usecase**:
`x-ms-parameterized-host` is used with enum, declaration:

```json
{
   "x-ms-parameterized-host": {
      "hostTemplate": "{geography}.atlas.microsoft.com",
      "parameters": [
        {
          "$ref": "#/parameters/GeographicResourceLocation"
        }
      ]
    },
   "parameters":  {
    "GeographicResourceLocation": {
      "name": "geography",
      "description": "This parameter specifies where the Azure Maps Creator resource is located.  Valid values are us and eu.",
      "in": "path",
      "required": true,
      "type": "string",
      "default": "us",
      "enum": [
        "us",
        "eu"
      ],
      "x-ms-enum": {
        "name": "GeographicResourceLocation",
        "modelAsString": true,
        "values": [
          {
            "value": "us",
            "description": "Used to access an Azure Maps Creator resource in the United States"
          },
          {
            "value": "eu",
            "description": "Used to access an Azure Maps Creator resource in Europe"
          }
        ]
      },
      "x-ms-parameter-location": "client"
    }
  }
}
```

Source: 
https://github.com/propheel/azure-rest-api-specs/blob/dev-maps-Microsoft.Maps-2.0-versioning-test/specification/maps/data-plane/Common/preview/1.0/common.json
https://github.com/propheel/azure-rest-api-specs/blob/dev-maps-Microsoft.Maps-2.0-versioning-test/specification/maps/data-plane/Render/preview/2.0/render.json

**Autorest invocation**:
```
autorest --go --track2 --use=@autorest/go@4.0.0-preview.22 specification/maps/data-plane/Creator/readme.md --go-sdk-folder=../azure-sdk-for-go --multiapi
AutoRest code generation utility [cli version: 3.2.3; node: v14.17.0, max-memory: 2048 MB]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
NOTE: AutoRest core version selected from configuration: 3.4.3.
   Loading AutoRest core      '/home/ambientlight/.autorest/@autorest_core@3.4.3/node_modules/@autorest/core/dist' (3.4.3)
INFORMATION: > Loading AutoRest extension '@autorest/go' (4.0.0-preview.22->4.0.0-preview.22)
INFORMATION: > Loading AutoRest extension '@autorest/modelerfour' (4.19.2->4.19.2)
Processing batch task - {"tag":"2.0-preview"} .
INFORMATION: > Loading AutoRest extension '@autorest/go' (4.0.0-preview.22->4.0.0-preview.22)
INFORMATION: > Loading AutoRest extension '@autorest/modelerfour' (4.19.2->4.19.2)

WARNING (PreCheck/CheckDuplicateSchemas): Checking for duplicate schemas, this could take a (long) while.  Run with --verbose for more detail.
[9.28 s] Generation Complete
```

**Golang:** 1.16.5
**Platform:** WSL2/Ubuntu 20.04

Autorest generates a valid [constants.go#L140](https://github.com/ambientlight/azure-sdk-for-go/blob/d97925c0dc464aa445862e5de620e41b80d389d5/services/preview/maps/2.0/creator/constants.go#L140-L145) however go complains on the assignment between string pointer and enum-backing type (Geography) pointer, unless explicit type cast is used.

```go
func newConnection(geography *Geography, cred azcore.Credential, options *connectionOptions) *connection {
	if options == nil {
		options = &connectionOptions{}
	}
	// .. reducted
	hostURL := "https://{geography}.atlas.microsoft.com"
	if geography == nil {
		defaultValue := "us"
		
		// cannot use &defaultValue (value of type *string) as *Geography value in assignmentcompilerIncompatibleAssign
		geography = &defaultValue
	}
	
	// cannot use *geography (variable of type Geography) as string value in argument to strings.ReplaceAll(IncompatibleAssign)
	hostURL = strings.ReplaceAll(hostURL, "{geography}", *geography)
	return &connection{u: hostURL, p: azcore.NewPipeline(options.HTTPClient, policies...)}
}
```

Reference: [connection.go](https://github.com/ambientlight/azure-sdk-for-go/blob/d97925c0dc464aa445862e5de620e41b80d389d5/services/preview/maps/2.0/creator/connection.go#L67-L71)

Should this be resolved by the addition of explicit casts?:

```go
geography = ((*Geography)(&defaultValue))

hostURL = strings.ReplaceAll(hostURL, "{geography}", (string)(*geography))
```

**To reproduce**:
```
git clone -b experiment-scripts https://github.com/ambientlight/azure-rest-api-specs.git
cd azure-rest-api-specs
./gen_go.sh
```

**Comment by ambientlight** [comment]
Also occurring for: 

https://github.com/ambientlight/azure-rest-api-specs/blob/5547eda058fa62078e929d746429f1c722755b93/specification/maps/data-plane/Route/preview/1.0/route.json#L303-L334

In 

https://github.com/ambientlight/azure-sdk-for-go/blob/d97925c0dc464aa445862e5de620e41b80d389d5/services/preview/maps/1.0/route/route_client.go#L141

**Comment by jhendrixMSFT** [comment]
Fixed in generator version 4.0.0-preview.23
